src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -884,6 +884,176 @@ internal class OnboardingSupervisor(
         }
     }
 
+    @Suppress("UnusedPrivateMember")
+    private fun retrieveSkipWithdrawalRouteExchange(
+        state: PerpetualState?,
+        decimals: Int,
+        gas: BigDecimal,
+        accountAddress: String,
+        sourceAddress: String,
+        subaccountNumber: Int?,
+    ) {
+        val usdcSize = helper.parser.asDecimal(state?.input?.transfer?.size?.usdcSize)
+        val fromAmount = if (usdcSize != null && usdcSize > gas) {
+            ((usdcSize - gas) * Numeric.decimal.TEN.pow(decimals)).toBigInteger()
+        } else {
+            null
+        }
+        val chainId = helper.environment.dydxChainId
+        val squidIntegratorId = helper.environment.squidIntegratorId
+        val dydxTokenDemon = helper.environment.tokens["usdc"]?.denom
+        val fromAmountString = helper.parser.asString(fromAmount)
+        val url = helper.configs.skipV2MsgsDirect()
+        val fromAddress = accountAddress
+        if (
+            fromAmount != null &&
+            fromAmount > 0 &&
+            fromAmountString != null &&
+            chainId != null &&
+            dydxTokenDemon != null &&
+            squidIntegratorId != null
+        ) {
```

Comment:
Nit: Can we add an `else` to call `Logger.e { }` to log an error?   Same with other functions.

Reply:
hmm wait do you mean if we don't have all the args? are we sure that's what we want? 

any transfer state change will fire off a retrieve deposit/withdrawal route function even if it doesn't make sense to do so, like when we clear the modal after submitting a withdraw/deposit. If we add an error log we'll be logging an error for all of those instances

Reply:
perhaps we should just `error()` out for things like `fromChain` and `nativeChainUSDCDenom` being null.